sdk/openai/openai/test/public/utils/asserts.ts

```diff
@@ -0,0 +1,487 @@
+  assert.isString(val.id);
+  assert.isBoolean(val.filtered);
+}
+
+function assertChoice(
+  choice: ChatCompletion.Choice | ChatCompletionChunk.Choice,
+  options: ChatCompletionTestOptions,
+): void {
+  const stream = options.stream;
+  if (stream) {
+    assertMessage((choice as ChatCompletionChunk.Choice).delta, options);
+    assert.isFalse("message" in choice);
+  } else {
+    assertMessage((choice as ChatCompletion.Choice).message, options);
+    assert.isFalse("delta" in choice);
+  }
+  assert.isNumber(choice.index);
+  ifDefined(choice.content_filter_results, assertContentFilterResultsForChoice);
+  ifDefined(choice.enhancements, assertAzureChatEnhancements);
+  ifDefined(choice.finish_details, assertChatFinishDetails);
+  ifDefined(choice.logprobs, assertLogProbability);
+  ifDefined(choice.finish_reason, assert.isString);
+}
+
+export async function assertCompletionsStream(
+  stream: AsyncIterable<Omit<OpenAI.Completions.Completion, "usage">>,
+  options: CompletionTestOptions = {},
+): Promise<number> {
+  return assertAsyncIterable(stream, (item) => assertCompletionsNoUsage(item, options));
+}
+
+async function assertAsyncIterable<T>(
+  val: AsyncIterable<T>,
+  validate: (x: T) => void,
+): Promise<number> {
+  const items: T[] = [];
+  for await (const item of val) {
+    try {
+      validate(item);
+    } catch (e: any) {
+      throw new Error(
+        `Error validating item:\n ${JSON.stringify(item, undefined, 2)}\n\n${e.message
+        }.\n\nPrevious items:\n\n${items
+          .map((x) => JSON.stringify(x, undefined, 2))
+          .join("\n")}\n\n Stack trace: ${e.stack}`,
+      );
+    }
+    items.push(item);
+  }
+  return items.length;
+}
+
+function assertLogprobs(logprobs: CompletionChoice.Logprobs): void {
+  ifDefined(logprobs.token_logprobs, (token_logprob) => assert.isNumber(token_logprob));
+  ifDefined(logprobs.text_offset, (text) => assert.isNumber(text));
+  ifDefined(logprobs.tokens, (token) => assert.isNumber(token));
+  ifDefined(logprobs.top_logprobs, (top_logprob) => assert.instanceOf(top_logprob, Object));
+}
+function assertLogProbability(
+  logProbability: OpenAI.Chat.Completions.ChatCompletion.Choice.Logprobs,
+): void {
+  assertNonEmptyArray(logProbability.content ?? [], assertTokenLogProbability);
+}
+
+function assertTokenLogProbability(tokenLogprob: ChatCompletionTokenLogprob): void {
+  ifDefined(tokenLogprob.bytes, (bytes) => {
+    assert.isNumber(bytes);
+  });
+  assert.isString(tokenLogprob.token);
+  assert.isNumber(tokenLogprob.logprob);
+}
+
+function assertUsage(usage: OpenAI.Completions.CompletionUsage | undefined): void {
+  assert.isDefined(usage);
+  const castUsage = usage as OpenAI.Completions.CompletionUsage;
+  assert.isNumber(castUsage.completion_tokens);
+  assert.isNumber(castUsage.prompt_tokens);
+  assert.isNumber(castUsage.total_tokens);
+}
+
+function assertIf(condition: boolean, val: any, check: (x: any) => void): void {
```

Comment:
Make the type a bit more rigorous?
```suggestion
function assertIf<T>(condition: boolean, val: T, check: (x: T) => void): void {
```